app/cdap/components/PipelineDiff/util/fetch.ts

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+import { MyPipelineApi } from 'api/pipeline';
+import { Observable } from 'rxjs/Observable';
+import { IPipelineConfig, IPipelineStage } from '../types';
+import { createPluginInfo, getArtifactKey, getKeyFromPluginProps } from '../util/helpers';
+
+type ApiCall<TOptions = any, TResponse = any> = (options: TOptions) => Observable<TResponse>;
```

Comment:
Why make this generic? 

is this better than just going `(options: IFetchPipelineOptions): Observable<IFetchPipelineResponse> =>` twice below?

Reply:
Because there is another function that makes an api call that has a different response (e.g.  `fetchExtraPluginProperties`)

Reply:
I understand that you do this twice - what I'm saying is that making this generic with `any` types actually makes this more complicated and less clear.

It would make more sense in my opinion to remove this generic type, and type it two separate times below because they're very simple functions.

Reply:
Oh I see. Yes it makes sense to just remove this type as it does make it less clear and more complicated than it needs to be, thanks!

Reply:
Updated code.